src/resource.cc

```diff
@@ -10,4 +10,47 @@ void Resource::BumpStateId() {
   nextstate_id_++;
 }
 
+double Resource::GetFillMass(Package::Ptr pkg) {
```

Comment:
I'm suddenly having second thoughts (third thoughts? fourth thoughts?) about moving this to `Resource`.  Here's why:
1. it only uses the `quantity()` from the `Resource` so we could pass that in a version that is `Package::GetFillMass(double qty)` and not need to worry about templates.
2. it depends on a lot of details of `Package` such as the fill strategies - separation of concerns suggests that it will be awkward to have to update a method in `Resource` if we change how `Package` internals work.

I was a little blinded by the templating challenge, and didn't see that these factors really make `GetFillMass` a property of a `Package`.  `PackageResource()` should stay in `Resource`.

Open for more discussion, sooner rather than later.